File: sms-api-java/authservice/src/main/java/com/example/mydemoproject/authservice/repository/UserRepository.java
package com.example.mydemoproject.authservice.repository;

import com.example.mydemoproject.authservice.model.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    @Query("select u from User u where u.username=:username")
    User findByUsername(@Param("username") String username);
}

File: sms-api-kotlin/userservice/src/main/kotlin/com/example/userservice/repository/UserRepository.kt
package com.example.userservice.repository

import com.example.userservice.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository

@Repository
interface UserRepository: JpaRepository<User, Long> {


    @Query("select u from User u where u.username=:username")
    fun findByUsername(@Param("username") username: String): User

}